ui/src/app/pages/analysis/notebook-redirect.spec.tsx

```diff
@@ -213,4 +214,37 @@ describe('NotebookRedirect', () => {
     expect(currentCardText(wrapper))
       .toContain(progressStrings.get(Progress.Redirecting));
   });
+
+  it('should navigate away after runtime transitions to deleting', async() => {
+    const navSpy = jest.fn();
+    NavStore.navigate = navSpy;
```

Comment:
Would this syntax work? Slightly easier to read IMO since its on one line: `jest.spyOn(NavStore, 'navigate')`

Reply:
NavStore is a bit odd - by default its methods are `undefined`... I think we may need to clean this up later for testing purposes. Unfortunately this doesn't work:

```
 Cannot spy the navigate property because it is not a function; undefined given instead

      217 | 
      218 |   it('should navigate away after runtime transitions to deleting', async() => {
    > 219 |     const navSpy = jest.spyOn(NavStore, 'navigate').mockImplementation(() => {});
```

There's also a problem with my current approach, which is that I never clear/restore the value of `navigate`. That said, it was undefined to begin with, so this is not the end of the world..